### PR TITLE
WIP: Update of the README document

### PR DESCRIPTION
**Pull request by quietlychris**
As the working group has grown a little bit, I figured it might be valuable to start thinking about a review of the layout and content of the Github organization. This pull request has a first draft of suggested changes to the README, which functionally acts as the WG's homepage for developers. Feedback and changes are requested!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,29 +5,31 @@
...
+- Help organize support for Rust libraries, tools, and projects in the machine learning domain
+- In rare instances, to assist with or assume the maintainership of projects important to the wider community
+
+## How can you contribute?
+
+The Rust-ML group has a Zulip chat. Feel free to take a look and say hi!
 - https://rust-ml.zulipchat.com
 
-### Organization
+We also host a video chats every two weeks. Meeting minutes can be found in the [Meetings](https://github.com/rust-ml/Meetings) repository.
```
Comment (deg4uss3r) [inline README.md]: I would also mention meetings are normally announced in the Zulip

File: README.md [hunk 2]
```diff
@@ -5,29 +5,31 @@
...
-- @[tiberiusferreira](https://github.com/tiberiusferreira)
-- @[xd009642](https://github.com/xd009642)
-- @[vadixidav](https://github.com/vadixidav)
+- [arewelearningyet.com](https://www.arewelearningyet.com/) 
+    - The Rust-ML group helps to maintain AreWeLearningYet by reviewing and accepting pull requests, and submitting new crates to the repository's lists
+- [linfa](https://github.com/rust-ml/linfa/)
+    - The Rust-ML group actively develops and maintiners linfa, a Rust machine learning framework/library, which fills a similar role as Python's scikit-learn
 
-We welcome interest from the community, so please feel free to participate!
+More importantly, Rust's machine learning community is filled of experimentation! Take a look at some of the projects currently being worked on by members on the working group's [Task Board](https://github.com/rust-ml/wg/blob/master/Task_Board.md), or take a look at some of the project's being discussed in our Zulip chat.
```
Comment (deg4uss3r) [inline README.md]: > More importantly, Rust's machine learning community is filled of experimentation! 

More importantly, Rust's machine learning community is filled with experimentation!

Comment (deg4uss3r) [inline README.md]: link Zulip chat again

File: README.md [hunk 3]
```diff
@@ -5,29 +5,31 @@
...
-### How can you contribute?
+## What do we do?
 
-Come say hello, read about, and weigh in on our [discussions](https://github.com/rust-ml/discussion)! We currently have several ongoing threads regarding upcoming scope of work, organizational structures, and more.
+In general, the Rust-ML working group's goals are to:
 
-The Rust-ML group has a Zulip channel. Feel free to take a look and say hi!
+- Create spaces where people interested in doing machine learning in Rust can meet, ask questions, and discuss their work
+- Help organize support for Rust libraries, tools, and projects in the machine learning domain
+- In rare instances, to assist with or assume the maintainership of projects important to the wider community
```
Comment (deg4uss3r) [inline README.md:16]: Maybe reword? I think I would like to foster Rust help to ML experts, the way this is worded might drive them off slightly?

Comment (xd009642) [reply README.md:16]: What about an additional bullet point, something like `- Create a mature production ready machine learning ecosystem`

File: README.md (hunk 1 above)
Comment (quietlychris) [reply README.md]: Good suggestion, will do

File: README.md (hunk 2 above)
Comment (quietlychris) [reply README.md]: Ugh, sorry about that. **with** experimentation, will change

Comment (quietlychris) [reply README.md]: Will do, can't have too many links to the chat...

File: README.md (hunk 3 above)
Comment (quietlychris) [reply README.md:16]: This is definitely the weakest area of the change, and I think it's worth some additional discussion. During the video meeting today, @deg4uss3r mentioned something along the lines of helping people with machine learning become familiar with Rust, and enabling people familiar with Rust to do machine learning. 

@xd009642 This is definitely an area for discussion about WG goals. For example, the Embedded Group very clearly directly develops crates and tooling, whereas the GameDev WG explicity does not build things like games/game engines, only fills the gaps in lower-level tools (https://github.com/rust-gamedev/wg/blob/master/charter.md). I'm unsure where along that spectrum we should sit, but I just trying to be wary of coming across in a way that says "hey, this WG has *the* way of doing machine learning in Rust".

Comment (xd009642) [reply README.md:16]: Hmm I'd say both working groups are working towards ecosystems just a bit more high level versus low level. What about "Begin to shape the developing rust machine learning ecosystem".

Comment (xd009642) [reply README.md:16]: Also I'll endeavour to turn up to the meeting next week as long as I have no clashes :smile:

Comment (quietlychris) [reply README.md:16]: Hmm, that might work. Let me make the other changes, and then we can maybe take this in particular to the woodshed a little bit more

Comment (quietlychris) [reply README.md:16]: I've made the other changes, but this is still an area that I think could use some more work

File: README.md [hunk 4]
```diff
@@ -5,29 +5,33 @@
 
 The Rust-ML working group is currently at an early stage of development, and is still deciding on several choices at an organizational level. This document will be updated accordingly as those decisions are made.
```
Comment (bytesnake) [inline README.md]: For me the `several choices` is a bit too vague. Perhaps we can word this as 

> The Rust-ML working group is currently at an early stage of development, and its prospect role and function in the wider Rust ecosystem is currently explored. This document will reflect those updates and inform the wider community on decisions made by the Rust-ML group.

I think it is useful to add a general statement, why we believe that Rust and Machine Learning is worthwhile:
> We strongly believe that Rust can contribute to the Machine Learning community with its correctness and speed. It brings the promise of a unified scientific workhorse by combining elegance and performance, previously only found in separate languages. We hope that this will produce libraries, which are easier to use, easier to iterate on and give a standardized level of performance across the ecosystem.

File: README.md [hunk 5]
```diff
@@ -5,29 +5,33 @@
...
 
-### How can you contribute?
+## What do we do?
 
-Come say hello, read about, and weigh in on our [discussions](https://github.com/rust-ml/discussion)! We currently have several ongoing threads regarding upcoming scope of work, organizational structures, and more.
+In general, the Rust-ML working group's goals are to:
 
-The Rust-ML group has a Zulip channel. Feel free to take a look and say hi!
+- Create spaces where people interested in doing machine learning in Rust can meet, ask questions, and discuss their work
```
Comment (bytesnake) [inline README.md:14]: Perhaps also:
 > Communicate interest in language features, like [GAT](https://github.com/rust-lang/rust/issues/44265), [const generics](https://github.com/rust-lang/rust/issues/44580), f16 types, which are important for easy-to-use machine learning libraries to the language team

File: README.md [hunk 6]
```diff
@@ -5,29 +5,33 @@
...
+- Help organize support for Rust libraries, tools, and projects in the machine learning domain
+- In rare instances, to assist with or assume the maintainership of projects important to the wider community
+
+We act as a resource to help people with experience in machine learning to become comfortable with Rust, enable people familiar with Rust to do machine learning, and provide guidance to people with neither. 
+
+## How can you contribute?
+
+The Rust-ML group has a Zulip chat. Feel free to take a look and say hi!
 - https://rust-ml.zulipchat.com
```
Comment (bytesnake) [inline README.md:24]: or if you already written something ask for a code review! :)

**Review by bytesnake — COMMENTED** [review]
I think that the intentions are clearly stated, I added something to the beginning and an idea for `goals`. (beware that I'm not a native speaker :sweat_smile:)

File: README.md (hunk 6 above)
Comment (quietlychris) [reply README.md:24]: Good point! It probably makes sense to have that explicitly laid out

File: README.md (hunk 5 above)
Comment (quietlychris) [reply README.md:14]: I'm a little hesitant to add this in unless there has been some direct contributions/advocacy for these languages features by members of the working group. Do you freely differently?

Comment (bytesnake) [reply README.md:14]: no, there is no example of direct contributions to the language (at least what I'm aware of)

File: README.md (hunk 4 above)
Comment (deg4uss3r) [reply README.md]: :+1: I really like these changes

Comment (quietlychris) [reply README.md]: Me too! I made a couple of edits (mostly grammar) to try to make it flow a little more, but hopefully it still communicates what you wanted. If not, we can change it back

**Review by deg4uss3r — APPROVED** [review]

**Comment by quietlychris** [comment]
Based on the comments, I'm going to merge these initial changes, with the understanding that there will probably be some revision moving forwards
